modules/packages/BLAS.chpl

```diff
@@ -2070,10 +2070,10 @@ module BLAS {
 
       :returns: Scalar value of dot product
   */
-  proc dot(X: [?D]?eltType,  Y: [D]eltType, incY: c_int = 1, incX: c_int = 1):eltType
-  where D.rank == 1 {
+  proc dot(X: [?xD]?eltType,  Y: [?yD] eltType, incY: c_int = 1, incX: c_int = 1):eltType
+  where xD: domain(1) && yD: domain(1) {
```

Comment:
``.rank == 1`` seems like a better fit than ``: domain(1)`` here.

Reply:
I agree. I think this is a remnant of my trial and error to avoid the ambiguous call error with `LinearAlgebra.Sparse.dot()`